src/service_server.cc

```diff
@@ -157,6 +157,63 @@ grpc::Status ServiceImpl::monitor(
 
   return grpc::Status::OK;
 }
+grpc::Status ServiceImpl::stream(
+    grpc::ServerContext *context, const chirp::StreamRequest *request,
+    grpc::ServerWriter<chirp::StreamReply> *writer) {
+  if (context == nullptr || request == nullptr || writer == nullptr) {
+    return grpc::Status(
+        grpc::FAILED_PRECONDITION,
+        "`ServerContext`, `StreamRequest`, or `writer` is nullptr.");
+  }
+
+  const int mseconds_per_wait = 50;
+  const int times_count = INT_MAX;
```

Comment:
This var and `cnt` below are unclear; I would recommend a more understandable variable name.